### ISSUE TITLE
$ false match in a regexp

### ISSUE THREAD

**Issue by ghost** [jfcherng/Sublime-ShellScriptImproved#16]
```
[[ "$str" =~ ^$'\t' ]]
              ^-------- matches keyword.control.anchors.regexp
                        instead of string.quoted.single.dollar.shell

[[ "$str" =~ ^abc$repl$ ]]
                 ^---------- matches keyword.control.anchors.regexp
                             instead of variable.other.normal.shell
```

The pattern undergoes variable expansion:

```
$ repl='def'; set -x; [[ 'abcdef' =~ ^abc$repl$ ]]; echo $?; set +x
+ [[ abcdef =~ ^abcdef$ ]]
+ echo 0
0
+ set +x
```

**Comment by jfcherng** [comment]
How does `^$'\t'` be interpreted in `[[ "$str" =~ ^$'\t' ]]`?

**Comment by ghost** [comment]
At the start, i.e. before `^` it is source.shell > meta.scope.logical-expression.shell > source.regexp > keyword.control.anchors.regexp

Before the `$` it matches the same.

After $ it matches source.shell > meta.scope.logical-expression.shell > source.regexp > meta.literal.regexp

`\t` sequence matches source.shell > meta.scope.logical-expression.shell > source.regexp > constant.character.escape.regexp which is right, I think.

The last single quote has the same scope as the opening one.

**Comment by jfcherng** [comment]
Thanks. Fixed by 01eb9ca0cdfef1ccbe5941269a979f2cc609c70c.
